intra/dialers/weird_split.go

```diff
@@ -0,0 +1,88 @@
+package dialers
+
+import (
+	"io"
+	"net"
+	"strings"
+	"syscall"
+)
+
+type WeirdSplitter struct {
+	*net.TCPConn
+	Used bool
+	Size int
+}
+
+func (s *WeirdSplitter) Write(b []byte) (int, error) {
+	conn := s.TCPConn
+	if s.Used {
+		// After the first write, there is no special write behavior.
+		return conn.Write(b)
+	}
+
+	// Setting `used` to true ensures that this code only runs once per socket.
+	s.Used = true
+	if len(b) < 2 {
+		return conn.Write(b)
+	}
```

Comment:
you think we should add a code comment on why `len(b) < 2` won't work?

(to my untrained eye, it isn't obvious)